topi/python/topi/x86/pooling.py

```diff
@@ -4,15 +4,41 @@
 from .. import generic
 from .. import tag
 
-def _parallel_sch(sch):
+def _parallel_sch(sch, oshape):
+    def vectorize(fused_axis, num_parallel_axis, vectorize_limit=64):
+        """Internal vectorization utility function."""
+        reorder_axis = [fused_axis]
+        for i in range(num_parallel_axis, len(sch.op.axis) - 1):
+            reorder_axis.append(sch.op.axis[i])
+        kw, kh = sch.op.reduce_axis
+        fuse_k = sch.fuse(kw, kh)
+        c = sch.op.axis[len(sch.op.axis) - 1]
+        reorder_axis += [fuse_k, c]
+        sch.reorder(*reorder_axis)
+        inner_length = oshape[len(oshape) - 1].value
+        if inner_length <= vectorize_limit:
+            sch.vectorize(c)
```

Comment:
looks like for NCHW, the vectorize over W, where padding applies, causes the problem you mentioned https://github.com/dmlc/tvm/issues/1516 . If this is the reason, >=5-D with innermost = H/W/h/w can also have problem?

Reply:
Yes. It will return LLVM error as well. Maybe we can turn off inline padding in this case?

Reply:
How about pass the layout from https://github.com/dmlc/tvm/blob/master/nnvm/python/nnvm/top/nn.py#L293 and check whether the HWhw are at the innermost, if so, just skip vectorize.

Reply:
Just curious, will the length be not power of 2 with padding or not a multiple of number of vector register lanes? If that is the case, we may want to do loop splitting and only apply vectorization the part with a multiple of the number of vector lanes.

Reply:
@zhiics I believe the crash with padding is due to the if...else inside the vectorize. x86 actually can deal with vectorize size which is not power of 2, not sure whether there is performance harm doing this. Yao does the splitting below.

Reply:
@zhiics Currently splitting factor is a factor of inner axis length no larger than splitting limit.